.github/workflows/pull_request.yml

```diff
@@ -1,32 +1,64 @@
 name: Pull request checks
+
 on:
   pull_request:
     push:
       branches:
         - main
 
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
+
 jobs:
-  verify_build:
+  build_and_test:
     runs-on: ubuntu-latest
     permissions:
       contents: read
       # Required to put a comment into the pull-request
       pull-requests: write
+
     steps:
-      - uses: actions/checkout@v3
+      - name: 📥 Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
 
-      - uses: oven-sh/setup-bun@v1
+      - name: 📥 Setup Bun
+        uses: oven-sh/setup-bun@v1
         with:
-          bun-version: 1.1.13
+          bun-version: 1.2.0
+
+      - name: 📥Install dependencies
+        # There is a bug with lockfile generation when using package overrides, so for now we
+        # install without the lockfile. :(
+        # run: bun install --frozen-lockfile
+        run: bun install
 
-      - name: Install dependencies
-        run: bun install --frozen-lockfile
+      - name: ⚡ Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
 
-      - name: Check formatting
+      - name: 📥 Install Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt clippy
+
+      - name: 📥 Install WASM Rust Toolchains
+        run: rustup target add wasm32-wasip1 wasm32-unknown-unknown
```

Comment:
you can under components do `targets wasm32-wasip1 wasm32-unknown-unknown` and that additionally installs both of those targets.